src/lib/OpenEXR/ImfBoxAttribute.h

```diff
@@ -14,43 +14,21 @@
 //
 //-----------------------------------------------------------------------------
 
-#include "ImfForward.h"
 #include "ImfExport.h"
-#include "ImfAttribute.h"
-#include "ImathBox.h"
 #include "ImfNamespace.h"
 
-OPENEXR_IMF_INTERNAL_NAMESPACE_HEADER_ENTER
+#include "ImfAttribute.h"
+#include <ImathBox.h>
 
+OPENEXR_IMF_INTERNAL_NAMESPACE_HEADER_ENTER
 
-typedef TypedAttribute<IMATH_NAMESPACE::Box2i> Box2iAttribute;
-
-template <>
-IMF_EXPORT
-const char *Box2iAttribute::staticTypeName ();
-template <>
-IMF_EXPORT
-void Box2iAttribute::writeValueTo (OPENEXR_IMF_INTERNAL_NAMESPACE::OStream &,
-                                   int) const;
-template <>
-IMF_EXPORT
-void Box2iAttribute::readValueFrom (OPENEXR_IMF_INTERNAL_NAMESPACE::IStream &,
-                                    int, int);
-
-
-typedef TypedAttribute<IMATH_NAMESPACE::Box2f> Box2fAttribute;
-template <>
-IMF_EXPORT
-const char *Box2fAttribute::staticTypeName ();
-template <>
-IMF_EXPORT
-void Box2fAttribute::writeValueTo (OPENEXR_IMF_INTERNAL_NAMESPACE::OStream &,
-                                   int) const;
-template <>
-IMF_EXPORT
-void Box2fAttribute::readValueFrom (OPENEXR_IMF_INTERNAL_NAMESPACE::IStream &,
-                                    int, int);
+using Box2iAttribute = TypedAttribute<IMATH_NAMESPACE::Box2i>;
+using Box2fAttribute = TypedAttribute<IMATH_NAMESPACE::Box2f>;
 
+#ifndef COMPILING_IMF_BOX_ATTRIBUTE
```

Comment:
This idiom occurs for every attribute type, it's worth explaining exactly what's going on here. Why do you need to avoid these declarations in the .cpp?